src/clients/GasFee.ts

```diff
@@ -0,0 +1,51 @@
+import { BN_ZERO } from "@carbon-sdk/util/number";
+import { SimpleMap } from "@carbon-sdk/util/type";
+import BigNumber from "bignumber.js";
+import { CarbonTx } from "..";
+
+class GasFee {
+    public readonly txGasCosts?: SimpleMap<BigNumber>;
+    public readonly txGasPrices?: SimpleMap<BigNumber>;
+
+    constructor(txGasCosts: SimpleMap<BigNumber>, txGasPrices: SimpleMap<BigNumber>) {
+        this.txGasCosts = txGasCosts
+        this.txGasPrices = txGasPrices
+    }
+
+    public static instance(txGasCosts?: SimpleMap<BigNumber>, txGasPrices?: SimpleMap<BigNumber>) {
+        if (!txGasCosts || !txGasPrices){
+            return new GasFee({},{})
+        }
+
+        return new GasFee(txGasCosts, txGasPrices);
+    }
+
+    public getFee(msgTypeUrl: string, denom: string): BigNumber {
```

Comment:
Think it would be good to add a default for denom.
```typescript
public getFee(msgTypeUrl: string, denom: string = DEFAULT_FEE_DENOM) {
  ...
}
```